backend/UserApp/views.py

```diff
@@ -150,36 +152,33 @@ def put(self, request, username):
 
 
 class UserUpdatePasswordView(APIView):
-    def put(self, request, username):
+    def put(self, request):
+        email = request.data.get('email', None)
```

Comment:
I don't think the frontend should have to pass you an email here. If the user is logged in, then you can get the ID for that user using this function I made [here](https://github.com/AnthonyOlijnyk/reservify/blob/53bdae41178939f284b787856606451940bdb89a/backend/core/utils/authorization.py#L28).

Since the frontend people wouldn't need to pass you an email / username anymore, you should communicate with them after you make the changes that [this](https://github.com/AnthonyOlijnyk/reservify/blob/53bdae41178939f284b787856606451940bdb89a/frontend/src/pages/UserDash.js#L94-L98) should be removed, and the request body [here](https://github.com/AnthonyOlijnyk/reservify/blob/53bdae41178939f284b787856606451940bdb89a/frontend/src/pages/UserDash.js#L114-L118) should be updated to match what you are expecting in your request.